src/main/resources/reference.conf

```diff
@@ -1358,6 +1358,103 @@ resourceTypes = {
     }
     reuseIds = true
   }
+  mct_mapping = {
+    actionPatterns = {
+      own = {
+        description = "perform owner actions on the resource"
+      }
+      delete = {
+        description = "delete this resource"
+      }
+      read_policies = {
+        description = "view all policies and policy details for this resource"
+      }
+      set_parent = {
+        description = "set the parent workspace of this resource"
+      }
+      read = {
+        description = "perform reader actions on the resource"
+      }
+      write = {
+        description = "perform writer actions on the resource"
+      }
+      "share_policy::owner" = {
+        description = "change the membership of the owner policy for this resource"
+      }
+      "share_policy::editor" = {
+        description = "change the membership of the editor policy for this resource"
+      }
+      "share_policy::writer" = {
+        description = "change the membership of the writer policy for this resource"
+      }
+      "share_policy::reader" = {
+        description = "change the membership of the reader policy for this resource"
+      }
+    }
+    ownerRoleName = "owner"
+    roles = {
+      owner = {
+        roleActions = ["delete", "read_policies", "share_policy::owner", "share_policy::editor", "share_policy::writer", "share_policy::reader", "own", "edit", "read", "write", "set_parent"]
+      }
+      editor = {
+        roleActions = ["delete", "edit", "read", "write"]
+      }
+      writer = {
+        roleActions = ["read", "write"]
+      }
+      reader = {
+        roleActions = ["read"]
+      }
+    }
+    reuseIds = false
+  }
+  mct_target = {
+    actionPatterns = {
+      own = {
+        description = "perform owner actions on the resource"
+      }
+      delete = {
+        description = "delete this resource"
+      }
+      read_policies = {
+        description = "view all policies and policy details for this resource"
+      }
+      read = {
+        description = "perform reader actions on the resource"
+      }
+      write = {
+        description = "perform writer actions on the resource"
+      }
+      "share_policy::owner" = {
+        description = "change the membership of the owner policy for this resource"
+      }
+      "share_policy::editor" = {
+        description = "change the membership of the editor policy for this resource"
+      }
+      "share_policy::writer" = {
+        description = "change the membership of the writer policy for this resource"
+      }
+      "share_policy::reader" = {
+        description = "change the membership of the reader policy for this resource"
+      }
+    }
+    ownerRoleName = "owner"
+    roles = {
+      owner = {
+        roleActions = ["delete", "read_policies", "share_policy::owner", "share_policy::editor", "share_policy::writer", "share_policy::reader", "own", "edit", "read", "write", "set_parent"]
+      }
+      editor = {
```

Comment:
Is it expected that only the owner is allowed to share?  I guess I'm not familiar with the use case but I would imagine that, for instance, an editor might want to share with readers?  Just as an example.

Reply:
that's a good call, I added that level of sharing (read-only) to the editor role

Reply:
editors being able to share with other editors is a viral sharing model, it is probably not what you want

Reply:
yup, that was dumb, thanks for calling it out. removed sharing for the editor role